src/os/bluestore/BlueStore.cc

```diff
@@ -480,21 +516,13 @@ void BlueStore::Cache::trim(uint64_t onode_max, uint64_t buffer_max)
   }
   while (buffer_size > buffer_max) {
     Buffer *b = &*i;
-    if (b->is_clean()) {
-      auto p = b->space->buffer_map.find(b->offset);
-      if (i != buffer_lru.begin()) {
-	--i;
-      }
-      dout(20) << __func__ << " rm " << *b << dendl;
-      b->space->_rm_buffer(p);
-    } else {
-      if (i != buffer_lru.begin()) {
-	--i;
-	continue;
-      } else {
-	break;
-      }
+    assert(b->is_clean());
+    auto p = b->space->buffer_map.find(b->offset);
+    if (i != buffer_lru.begin()) {
+      --i;
     }
+    dout(20) << __func__ << " rm " << *b << dendl;
+    b->space->_rm_buffer(p);
```

Comment:
Looks like one can have a  BufferSpace::_rm_buffer(offset) instead of doing find + _rm_buffer(iterator). There are several similar snippets below